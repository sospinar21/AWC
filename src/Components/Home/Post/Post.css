.post-container {
  width: 100%;
  height: 90%;
  overflow-y: scroll; }

.post {
  width: 90%;
  background-color: #ffffff;
  margin: auto;
  height: 130px;
  margin: 20px auto;
  box-shadow: 0px 1px 2px lightgrey;
  padding: 10.5px;
  box-sizing: border-box;
  display: flex;
  cursor: pointer; }

button:disabled {
  background-color: #b4b4b4;
  cursor: default; }

.post:hover {
  background-color: #d3d3d33b; }

.post-img {
  border-radius: 5px;
  -webkit-box-shadow: 1px 2px 3px #ccc;
  box-shadow: 1px 2px 3px #ccc;
  width: 20%;
  height: 100%;
  background-image: url("../../../assets/event.png");
  background-size: cover; }
  .post-img img {
    width: 100%;
    height: 100%; }

.post-description {
  padding: 10px;
  width: 80%;
  box-sizing: border-box; }
  .post-description h3 {
    font-size: 17px;
    margin: 10px;
    margin: 10px;
    color: #b62d4c; }

.text-description {
  margin: 0 10px 15px;
  font-size: 18 px;
  overflow-y: scroll;
  height: 60px; }
  .text-description p {
    letter-spacing: .35px;
    line-height: 1.2; }

.likes {
  display: flex;
  margin-top: -20px;
  float: right; }
  .likes h3 {
    margin-right: 25px; }

.icon {
  line-height: 0;
  margin: 20px; }

.post-area {
  margin: 14px 50px;
  border-radius: 2px; }

.post-menu {
  background: #F3F3F3;
  display: flex;
  margin: 0 0px -9px 0px;
  text-align: right;
  color: #C0C0C0;
  font-size: 11px;
  box-sizing: border-box; }
  .post-menu li {
    cursor: pointer;
    text-align: center;
    list-style: none;
    outline: none;
    border: none;
    border-left: 1px solid lightgrey;
    width: 20%;
    padding: 6px;
    font-size: 14px;
    background-color: #f3f3f3; }

.label {
  margin-top: -19px;
  display: block;
  float: left;
  background: #FFFFFF;
  height: 14px;
  padding: 2px 5px 2px 5px;
  color: #B9B9B9;
  font-size: 14px; }

.post-box {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  width: 100%;
  display: block;
  outline: none;
  border: none;
  height: 72px;
  line-height: 25px;
  font-size: 16px;
  padding: 0; }

.submit-post {
  cursor: pointer;
  outline: none;
  font-size: 14px;
  margin-top: 13px;
  background: #b62d4c;
  border: none;
  padding: 10px 20px 10px 20px;
  border-bottom: 3px solid #b62d4c;
  border-radius: 3px;
  color: #ffffff;
  width: 100%; }

.post-cont {
  display: block;
  padding: 9px;
  border: 1px solid #DDDDDD;
  margin-bottom: 30px;
  border-radius: 3px; }
