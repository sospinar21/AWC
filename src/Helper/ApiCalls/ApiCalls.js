import React, {Component} from 'react';

class ApiCalls extends Component {
  constructor(){
    super()
  }

  fetchEvents = async () => {
    const url = `https://api.awc.dance/events?city=vermont`;
    const response = await fetch(url)
    const data= await response.json();
    return data;
  } 

  fetchStudios = () => {
    const url = `https://api.awc.dance/?city=80202`;
    return fetch(url, {
      method: 'GET',
      headers: {
        'Content-type': 'application/json',
      }
    }).then( response => response.json() )
      .then( data => data )   
  }

  fetchVideos= async () => {
    var url = 'https://api.awc.dance/youtubevideos'
    const response = await fetch(url)
    const data = await response.json()
    return data
  }
}

export default ApiCalls;