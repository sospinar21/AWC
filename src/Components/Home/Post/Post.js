import React, { Component } from 'react';
import { connect } from 'react-redux';
import ApiCalls from '../../../Helper/ApiCalls/ApiCalls'
import './Post.css';

export class Post extends Component {
  constructor(props) {
    super(props) 

    this.state = {
      category: '',
      input: '',
      likes: 0,
      dislikes: 0
    }
  }

  getPost = (e) => {
    const input = e.target.value
    this.setState({input})
  }

  sendPost = (e) => {
    e.preventDefault()
    this.props.addPost(this.state)
    this.sendToLambda() 
    this.setState({category:'', input:''})
  }

  listenClick = (e) => {
    const selected = e.target.closest('li').innerText;
    this.setState({
      category : selected
    })
  }

  sendToLambda = async () => {
    const api = new ApiCalls()
    const {token, user} = this.props.user
    const {category, input} = this.state
    const response = await api.postComment(token, user, input, category)
    console.log(response)
  }
  
  displayPostBox = () => {
    return (
      <div className='post-area'>
        <form className='post-cont'> 
          <label className='label' htmlFor= 'post'>Post here!</label>
          <textarea name='post' className='post-box' value={this.state.input} onChange={(e) => this.getPost(e)}/>
          <div className='post-menu' onClick={(e) => this.listenClick(e)}>
            <li>#Image</li>
            <li>#Video</li>
            <li>#Network </li>
            <li>#Pro</li>
            <li>#Entertainment</li>
          </div> 
          <button type='submit' className='submit-post' onClick={(e) => this.sendPost(e)}> Submit </button>
        </form>
      </div>
    )
  }


  render () {
    return (
      <div className='form-container'>
        {this.displayPostBox()}
      </div>
    );
  }
}


export const mapStateToProps = (state) => {
  return ({
    user: state.user
  });
};

export const mapDispatchToProps = dispatch => ({
});

Post.propTypes = {

};

export default connect(mapStateToProps, mapDispatchToProps)(Post);