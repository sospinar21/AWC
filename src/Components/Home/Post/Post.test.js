import {Post, mapStateToProps, mapDispatchToProps} from '../Post/Post'
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import ApiCalls from '../../../Helper/ApiCalls/ApiCalls';

describe('Post', () => {
  let post;
  let mockProps;
  let event;
  const api = new ApiCalls();

  beforeEach(() => {
    mockProps = {
      addUser: jest.fn(),
      user: {username:'Laura'},
      addPost: jest.fn()
    };

    post = shallow(<Post {...mockProps}/>, { disableLifecycleMethods: true });

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve({data: {}})
    }));

    event = {
      target: {
        name: 'email',
        value: 'user@user.com'
      },
      preventDefault: () => {}
    };

  });

  it('matches snapshot', () => {
    let post = shallow(<Post {...mockProps}/>, { disableLifecycleMethods: true });
    expect(post).toMatchSnapshot();
  });

  describe('getPost', () => {

    it('sets the state with the selected location', () => {
      
      event = {
        target: {
          name: 'content',
          value: 'Dancer for ever'
        },
        preventDefault: () => {}
      };

      post.instance().getPost(event);

      expect(post.state().content).toEqual('Dancer for ever');
    });
  });

  describe('sendPost', () => {

    it('calls props.addPost', () => {
      
      post.instance().sendPost(event);

      expect(mockProps.addPost).toHaveBeenCalled()
    });

    it('sets the state to default', () => {
      
      post.state().content = 'Dancer for ever'

      post.instance().sendPost(event);

      expect(post.state().content).toEqual('');
    });
  });

  describe('sendToLambda', () => {
  
    it('calls fetchGoogle with the correct params', async () => {
      
      const expected = 'https://api.awc.dance/postcomment?name=Laura&content=&type=Post'

      post.instance().sendToLambda();

      expect(window.fetch).toHaveBeenCalledWith(expected)
    });
  });

  describe('validatePost', () => {

    it('returns true if the input is empty', () => {
      
      post.state().content = ''

      const validate = post.instance().validatePost();

      expect(validate).toEqual(true);
    });
  });

  describe('mapStateToProps', () => {

    let mappedProps

    it('returns an object with the user', () => {

      const mockState = {
        user: {username: 'Kai'}
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(mockState);
      
    })
  })

  describe('mapDispatchToProps', () => {

    it('should call dispatch with the correct params', () => {
    
      let mockDispatch = jest.fn();
      let mappedProps = mapDispatchToProps(mockDispatch);
      
      const user = {name:'Steph'};      
      const mockAction = {
        type: 'ADD_USER',
        user
      };
  
      mappedProps.addUser(user);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });

  });

});