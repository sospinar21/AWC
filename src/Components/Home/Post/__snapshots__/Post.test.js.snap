// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Post
    addPost={[Function]}
    addUser={[Function]}
    user={
        Object {
            "username": "Laura",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="post-area"
>
        <form
                className="post-cont"
        >
                <label
                        className="label"
                        htmlFor="post"
                >
                        Post here!
                </label>
                <textarea
                        className="post-box"
                        name="post"
                        onChange={[Function]}
                        value=""
                />
                <button
                        className="submit-post"
                        disabled={true}
                        onClick={[Function]}
                        type="submit"
                >
                         Submit 
                </button>
        </form>
</div>,
      "className": "form-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="post-cont"
>
          <label
                    className="label"
                    htmlFor="post"
          >
                    Post here!
          </label>
          <textarea
                    className="post-box"
                    name="post"
                    onChange={[Function]}
                    value=""
          />
          <button
                    className="submit-post"
                    disabled={true}
                    onClick={[Function]}
                    type="submit"
          >
                     Submit 
          </button>
</form>,
        "className": "post-area",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="label"
              htmlFor="post"
>
              Post here!
</label>,
            <textarea
              className="post-box"
              name="post"
              onChange={[Function]}
              value=""
/>,
            <button
              className="submit-post"
              disabled={true}
              onClick={[Function]}
              type="submit"
>
               Submit 
</button>,
          ],
          "className": "post-cont",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Post here!",
              "className": "label",
              "htmlFor": "post",
            },
            "ref": null,
            "rendered": "Post here!",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "post-box",
              "name": "post",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " Submit ",
              "className": "submit-post",
              "disabled": true,
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": " Submit ",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="post-area"
>
          <form
                    className="post-cont"
          >
                    <label
                              className="label"
                              htmlFor="post"
                    >
                              Post here!
                    </label>
                    <textarea
                              className="post-box"
                              name="post"
                              onChange={[Function]}
                              value=""
                    />
                    <button
                              className="submit-post"
                              disabled={true}
                              onClick={[Function]}
                              type="submit"
                    >
                               Submit 
                    </button>
          </form>
</div>,
        "className": "form-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            className="post-cont"
>
            <label
                        className="label"
                        htmlFor="post"
            >
                        Post here!
            </label>
            <textarea
                        className="post-box"
                        name="post"
                        onChange={[Function]}
                        value=""
            />
            <button
                        className="submit-post"
                        disabled={true}
                        onClick={[Function]}
                        type="submit"
            >
                         Submit 
            </button>
</form>,
          "className": "post-area",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="label"
                htmlFor="post"
>
                Post here!
</label>,
              <textarea
                className="post-box"
                name="post"
                onChange={[Function]}
                value=""
/>,
              <button
                className="submit-post"
                disabled={true}
                onClick={[Function]}
                type="submit"
>
                 Submit 
</button>,
            ],
            "className": "post-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Post here!",
                "className": "label",
                "htmlFor": "post",
              },
              "ref": null,
              "rendered": "Post here!",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "post-box",
                "name": "post",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " Submit ",
                "className": "submit-post",
                "disabled": true,
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": " Submit ",
              "type": "button",
            },
          ],
          "type": "form",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
